pkg/cluster/internal/create/kubeadm-init.go

```diff
@@ -155,6 +154,29 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
 	return nil
 }
 
+func getHostPort(ec *execContext, controlPlane *nodes.Node) (int, error) {
+	// in single control-plane cluster, the node exposing the API server to the outside world is the control-plane itself
+	node := controlPlane
+	port := kubeadm.APIServerPort
+
+	// in case an lb is defined, use the lb node instead
+	if ec.ExternalLoadBalancer() != nil {
+		loadBalancerHandle, ok := ec.NodeFor(ec.ExternalLoadBalancer())
+		if !ok {
+			return 0, errors.Errorf("unable to get the handle for operating on node: %s", ec.ExternalLoadBalancer().Name)
+		}
+
+		node = loadBalancerHandle
+		port = haproxy.ControlPlanePort
```

Comment:
with respect to debugging e2e, would it make sense to add a `log.Info` message to indicate that we are using the LB port?

Reply:
done